sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -30,17 +30,28 @@ template <class To, class From> To cast(From value) {
   return (To)(value);
 }
 
+// Older versions of GCC don't like "const" here
+#if defined(__GNUC__) && (__GNUC__ < 7 || (__GNU__C == 7 && __GNUC_MINOR__ < 2))
+#define CONSTFIX constexpr
+#else
+#define CONSTFIX const
+#endif
+
 // Names of USM functions that are queried from OpenCL
-const char clHostMemAllocName[] = "clHostMemAllocINTEL";
-const char clDeviceMemAllocName[] = "clDeviceMemAllocINTEL";
-const char clSharedMemAllocName[] = "clSharedMemAllocINTEL";
-const char clMemFreeName[] = "clMemFreeINTEL";
-const char clSetKernelArgMemPointerName[] = "clSetKernelArgMemPointerINTEL";
-const char clEnqueueMemsetName[] = "clEnqueueMemsetINTEL";
-const char clEnqueueMemcpyName[] = "clEnqueueMemcpyINTEL";
-const char clEnqueueMigrateMemName[] = "clEnqueueMigrateMemINTEL";
-const char clEnqueueMemAdviseName[] = "clEnqueueMemAdviseINTEL";
-const char clGetMemAllocInfoName[] = "clGetMemAllocInfoINTEL";
+CONSTFIX char clHostMemAllocName[] = "clHostMemAllocINTEL";
+CONSTFIX char clDeviceMemAllocName[] = "clDeviceMemAllocINTEL";
+CONSTFIX char clSharedMemAllocName[] = "clSharedMemAllocINTEL";
+CONSTFIX char clMemFreeName[] = "clMemFreeINTEL";
+CONSTFIX char clSetKernelArgMemPointerName[] = "clSetKernelArgMemPointerINTEL";
+CONSTFIX char clEnqueueMemsetName[] = "clEnqueueMemsetINTEL";
+CONSTFIX char clEnqueueMemcpyName[] = "clEnqueueMemcpyINTEL";
+CONSTFIX char clEnqueueMigrateMemName[] = "clEnqueueMigrateMemINTEL";
+CONSTFIX char clEnqueueMemAdviseName[] = "clEnqueueMemAdviseINTEL";
+CONSTFIX char clGetMemAllocInfoName[] = "clGetMemAllocInfoINTEL";
```

Comment:
I was actually thinking to just
```
auto constexpr clHostMemAllocName = "clHostMemAllocINTEL";
```
but it might not work on some old compilers... Who cares about old compilers? SYCL is about the future. :-)